### PR TITLE
ARROW-5646: [Crossbow][Documentation] Move the user guide to the Sphinx documentation

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by kszucs** [comment]
@nealrichardson please review

File: docs/source/developers/crossbow.rst [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+   -->
+
+Packaging and Testing with Crossbow
+===================================
+
+The content of this directory aims for automating the historically error
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: Which directory? and remove "historically error prone"

File: docs/source/developers/crossbow.rst [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+prone process of Arrow packaging and integration testing.
+
+Packages:
+  - C++ and Python `conda-forge packages`_ for Linux, Mac and Windows
+  - Python `Wheels`_ for Linux, Mac and Windows
+  - C++ and GLib `Linux packages`_ for multiple distributions
+  - Java for Gandiva
+
+Integration tests:
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: Add links to the code/READMEs for these

File: docs/source/developers/crossbow.rst [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+~~~~~~~~~
+
+Individual jobs are executed on public CI services, currently: - Linux:
+TravisCI - Mac: TravisCI - Windows: AppVeyor
+
+Queue
+~~~~~
+
+Because of the nature of how the CI services work, the scheduling of
+jobs happens through an additional git repository, which acts like a job
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: Where is this other git repository?

File: docs/source/developers/crossbow.rst [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+   -  turn off Travis’ `auto cancellation`_ feature on branches
+
+3. Clone the newly created, by default the scripts looks for
+   ``crossbow`` next to arrow repository.
+
+   .. code:: bash
+
+      git clone https://github.com/<user>/crossbow crossbow
+
+   **Important note:** Crossbow only supports GitHub token based
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: I worked out how to use GitHub deploy keys for arrow-site (though INFRA is blocking their use) so we could add that here, if there's anyone who's uncomfortable using a PAT with broad scope.

File: docs/source/developers/crossbow.rst [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+   -  TravisCI: ``https://travis-ci.org/<ghuser>/<ghrepo>/settings``
+   -  Appveyor:
+      ``https://ci.appveyor.com/project/<ghuser>/<ghrepo>/settings/environment``
+   -  CircleCI:
+      ``https://circleci.com/gh/<ghuser>/<ghrepo>/edit#env-vars``
+
+   On Appveyor check the ``skip branches without appveyor.yml`` checkbox
+   on the web UI under crossbow repository’s settings.
+
+7. Install Python 3.6:
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: Exactly 3.6? Doesn't work on 3.7?

File: docs/source/developers/crossbow.rst [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+
+   Miniconda is preferred, see installation instructions:
+   https://conda.io/docs/user-guide/install/index.html
+
+8. Install the python dependencies for the script:
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: There's not a better way to specify requirements?

File: docs/source/developers/crossbow.rst [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+   server can be used.
+
+1. `Create the queue repository`_
+
+2. Enable `TravisCI`_, `Appveyor`_, `Azure Pipelines_` and `CircleCI`_
+   integrations on for the newly created queue repository.
+
+   -  turn off Travis’ `auto cancellation`_ feature on branches
+
+3. Clone the newly created, by default the scripts looks for
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: So the "queue repository" I have to create must be called "crossbow"? That should be more explicit.

File: docs/source/developers/crossbow.rst [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+
+      $ git clone https://github.com/kszucs/arrow
+      $ git clone https://github.com/kszucs/crossbow
+
+      $ cd arrow/dev/tasks
+      $ python crossbow.py submit conda-win conda-linux conda-osx
+
+2. Gets the HEAD commit of the currently checked out branch and
+   generates the version number based on `setuptools_scm`_. So to build
+   a particular branch, just check out before running the script:
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: Remove "just" since it's not "just" checking out, according to the "Note" that follows.

Comment (nealrichardson) [reply docs/source/developers/crossbow.rst]: And this "how to" should go somewhere else. This section IIUC is explaining what crossbow does when you call it, not how to control it.

File: docs/source/developers/crossbow.rst [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+   snippet will build kszucs’s fork instead of the upstream apache/arrow
+   repository.
+
+   .. code:: bash
+
+      $ git clone https://github.com/kszucs/arrow
+      $ git clone https://github.com/kszucs/crossbow
+
+      $ cd arrow/dev/tasks
+      $ python crossbow.py submit conda-win conda-linux conda-osx
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: How do I know what arguments to pass to crossbow?

File: docs/source/developers/crossbow.rst [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+5. Pushes the modified branches to GitHub which triggers the builds. For
+   authentication it uses GitHub OAuth tokens described in the install
+   section.
+
+Query the build status
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: bash
+
+   python crossbow.py status <build id / branch name>
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: Where do I get the build id / branch name from?

File: docs/source/developers/crossbow.rst [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: bash
+
+   python crossbow.py artifacts <build id / branch name>
+
+Examples
+~~~~~~~~
+
+The script accepts a pattern as a first argument to narrow the build
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: Looks like this is missing its example. No idea what "pattern" or "first argument" means here.

File: docs/source/developers/crossbow.rst [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+
+.. code:: bash
+
+   $ python crossbow.py submit --dry-run task_name
+
+Run only ``conda`` package builds and a Linux one:
+
+.. code:: bash
+
+   $ python crossbow.py submit -g conda centos-7
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: What's -g?

File: docs/source/developers/crossbow.rst [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
+
+Run ``wheel`` builds:
+
+.. code:: bash
+
+   $ python crossbow.py submit --group wheel
+
+In order to submit docker tests defined in the ``docker-compose.yml`` use the
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: Where is docker-compose.yml?

**Review by nealrichardson — CHANGES_REQUESTED** [review]

File: docs/source/developers/crossbow.rst [hunk 14]
```diff
@@ -0,0 +1,284 @@
...
+  - Spark
+
+Architecture
+------------
+
+Executors
+~~~~~~~~~
+
+Individual jobs are executed on public CI services, currently: - Linux:
+TravisCI - Mac: TravisCI - Windows: AppVeyor
```
Comment (nealrichardson) [inline docs/source/developers/crossbow.rst]: It appears that we're actually using Circle CI? https://github.com/ursa-labs/crossbow/branches/all?page=2&query=nightly-625&utf8=%E2%9C%93

File: docs/source/developers/crossbow.rst (hunk 12 above)
Comment (kszucs) [reply docs/source/developers/crossbow.rst]: Renamed to the long name of the option `--group`

File: docs/source/developers/crossbow.rst (hunk 10 above)
Comment (kszucs) [reply docs/source/developers/crossbow.rst]: Added a note above.

File: docs/source/developers/crossbow.rst (hunk 7 above)
Comment (kszucs) [reply docs/source/developers/crossbow.rst]: No, that's just the default. I've tried to express it better.

**Comment by nealrichardson** [comment]
+1. I still have questions but I won't let the perfect be the enemy of the good.

**Review by nealrichardson — APPROVED** [review]

**Pull request closed without merging**